web/html/src/manager/admin/list-payg/list-payg.tsx

```diff
@@ -0,0 +1,205 @@
+import { hot } from "react-hot-loader/root";
+
+import * as React from "react";
+import { useEffect } from "react";
+
+import _truncate from "lodash/truncate";
+
+import PaygStatus from "manager/admin/payg-shared/common/payg-status";
+
+import { Button } from "components/buttons";
+import { FromNow } from "components/datetime/FromNow";
+import withPageWrapper from "components/general/with-page-wrapper";
+import { ServerMessageType } from "components/messages";
+import { SectionToolbar } from "components/section-toolbar/section-toolbar";
+import { Column } from "components/table/Column";
+import { SearchField } from "components/table/SearchField";
+import { Table } from "components/table/Table";
+import { showSuccessToastr } from "components/toastr/toastr";
+import { HelpLink } from "components/utils";
+
+import { Utils } from "utils/functions";
+
+const _SETUP_WIZARD_STEPS = [
+  {
+    id: "wizard-step-proxy",
+    label: "HTTP Proxy",
+    url: "/rhn/admin/setup/ProxySettings.do",
+    active: false,
+  },
+  {
+    id: "wizard-step-credentials",
+    label: "Organization Credentials",
+    url: "/rhn/admin/setup/MirrorCredentials.do",
+    active: false,
+  },
+  {
+    id: "wizard-step-suse-products",
+    label: "Products",
+    url: "/rhn/manager/admin/setup/products",
+    active: false,
+  },
+  {
+    id: "wizard-step-suse-payg",
+    label: "Pay-as-you-go",
+    url: "/rhn/manager/admin/setup/payg",
+    active: true,
+  },
+];
+
+type PaygOverviewType = {
+  id: String;
+  host: String;
+  description: String;
+  status: String;
+  statusMessage: String;
+  lastChange: moment.Moment;
+};
+
+type Props = {
+  payg_instances: Array<PaygOverviewType>;
+  flashMessage?: ServerMessageType;
+};
+const ListPayg = (props: Props) => {
+  useEffect(() => {
+    if (props.flashMessage) {
+      showSuccessToastr(props.flashMessage);
+    }
+  }, []);
+
+  const searchData = (row, criteria) => {
+    const keysToSearch = ["host", "description", "status"];
+    if (criteria) {
+      return keysToSearch
+        .map((key) => row[key])
+        .join()
```

Comment:
If I'm not mistaken, this will give unexpected results when someone searches for a string containing a comma etc. Perhaps something along these lines would work instead?  

```ts
if (criteria) {
  const needle = criteria.toLocaleLowerCase();
  return keysToSearch.map(...).some(item => item.includes(needle));
}
```

Reply:
Done. Thanks for the int, it's way better now.